### PR TITLE
XDP UT and fixes to the XDP program

### PR DESCRIPTION
**Pull request by mazdakn**
## Description

This PR includes the following related to XDP:
- Some changes to support running XDP UT
- Some XDP UT test cases
- Some fixes to the XDP program and shared functions with TC

## Todos
- [x] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done

## Release Note

```release-note
None required
```

### CODE REVIEW

File: bpf/ut/xdp_test.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+type xdpTest struct {
+	Description string
+	Rules       *polprog.Rules
+	IPv4Header  *layers.IPv4
+	NextHeader  gopacket.Layer
+	Drop        bool
+	Metadata    bool
+}
+
+// TODO: Add some test cases to match destination IP:Port
```
Comment (neiljerram) [inline bpf/ut/xdp_test.go:105]: Can we do this now?

**Review by neiljerram — CHANGES_REQUESTED** [review]
Just a couple of things, plus wondering if we can add dest IP cases now?

File: bpf/ut/xdp_test.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2019-2021 Tigera, Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (mazdakn) [reply bpf/ut/xdp_test.go:1]: right, justed pasted it from another file. Will fix it.

File: bpf/ut/xdp_test.go (hunk 1 above)
Comment (mazdakn) [reply bpf/ut/xdp_test.go:105]: Will check it.

Comment (mazdakn) [reply bpf/ut/xdp_test.go]: it works.

**Review by neiljerram — APPROVED** [review]
LGTM, thanks!

**Pull request merged**